### PR TITLE
Correct Swift Decimal.divide to call NSDecimalDivide.

### PR DESCRIPTION
**Pull request by mjmdavis**
#### What's in this pull request?
- Correct Swift Decimal.divide to call NSDecimalDivide.
- Test case for Decimal.divide and Decimal.multiply.

I'm not sure that this is the best place or way to perform this test but I'm also not entirely clear on what the existing tests are testing.
#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

None

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

A smoke test on macOS does the following:
1. Builds the compiler incrementally.
2. Builds the standard library only for macOS. Simulator standard libraries and
   device standard libraries are not built.
3. lldb is not built.
4. The test and validation-test targets are run only for macOS. The optimized
   version of these tests are not run.

A smoke test on Linux does the following:
1. Builds the compiler incrementally.
2. Builds the standard library incrementally.
3. lldb is built incrementally.
4. The swift test and validation-test targets are run. The optimized version of these
   tests are not run.
5. lldb is tested.

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

File: stdlib/public/SDK/Foundation/Decimal.swift [hunk 1]
```diff
@@ -73,7 +73,7 @@ extension Decimal {
 
     public mutating func divide(by other: Decimal) {
         var rhs = other
-        NSDecimalMultiply(&self, &self, &rhs, .plain)
```
Comment (mjmdavis) [inline stdlib/public/SDK/Foundation/Decimal.swift]: I assume that this was some kind of copy/paste error.

**Comment by mjmdavis** [comment]
Probably worth mentioning that I came across this bug in XCode `Version 8.0 beta 3 (8S174q)`. I thought that I might have lost it for a second there but... I'm still apparently mostly sane 😝 .

**Comment by parkera** [comment]
Thanks @mjmdavis !

**Comment by parkera** [comment]
@swift-ci Please test and merge

**Comment by parkera** [comment]
The OS X test passed and this code is not present on Linux, so I'm going to merge.

**Pull request merged**